x-pack/plugins/enterprise_search/server/lib/enterprise_search_request_handler.ts

```diff
@@ -188,14 +210,26 @@ export class EnterpriseSearchRequestHandler {
     this.log.error(errorMessage);
     if (e instanceof Error) this.log.debug(e.stack as string);
 
-    return response.customError({ statusCode: 502, body: errorMessage });
+    return response.customError({ statusCode: 502, headers: this.headers, body: errorMessage });
```

Comment:
I went back and forth on whether repeating `headers: this.headers` in each response handler felt too un-DRY - eventually I just left it alone for flexibility (on the off-chance a specific response needed to pass its own custom header in the future? 🤷)

That being said I can definitely poke at it/refactor more if it feels too repetitive, just let me know!